### PR TITLE
Use the new way to obtain tokens and codes via iaa:true

### PR DESCRIPTION
**Pull request by bakulf**

### CODE REVIEW

File: src/authenticationinapp/authenticationinapplistener.cpp
Comment (birdsarah) [inline src/authenticationinapp/authenticationinapplistener.cpp:263]: I would have thought you needed, at least, scope. I was expecting you to need all of state, scope, accessType.

Just because the request works without these three doesn't mean it actually works:
* accessType, I thought, indicates we need a long-term refreshToken
* scopes are all the scopes we need for the support form, profile checking (including subscription checking), and in-app subscritpion to work.
* state is an important part of the integrity of the oauth flow (@jswinarton will have the details of what we need when on that bit)

**Review by birdsarah — COMMENTED** [review]
I think this looks broadly right. I haven't done the most thorough job so apologies if my comment about scopes is out of place. 

I don't think either jeremy or i have context on the different types of requests you have to make for different login scenarios as we've just handed that all off to fxa. You may want to get an fxa engineer to review this.

But in terms of passing things around. The back end looks good i.e. you use the redirect url you get from fxa and redirect to it and get a code back. And the front looks good - where you get an object from us - although i'll let jeremy approve or not as he has the details on the apis.

File: src/authenticationinapp/authenticationinapplistener.cpp
Comment (bakulf) [reply src/authenticationinapp/authenticationinapplistener.cpp:263]: All is good here. See https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/docs/api.md#post-accountcreate
The account creation doesn't need accessType, scopes, or state. Those are required only for the final step: `POST /oauth/authorization`

**Comment by bakulf** [comment]
I haven't changed the use of the FxA API. I don't think I need a review from the FxA team. This review request is mainly about:
- code syntax, style, etc.
- interaction with guardian
- the temporary fix for the unit-test (no JSON for errors)

**Review by jswinarton — APPROVED** [review]

File: src/networkrequest.cpp
Comment (bakulf) [reply src/networkrequest.cpp:554]: `/v1/session/verify_code` is a non-documented FxA API that wants the scopes as an array. I had to revert-engineer what the web page does to see the correct params of this endpoint.
I already spoke with the FxA team about this missing documentation, but it's better to file a bug. So far, this code works.

**Pull request merged**
